frontend/javascripts/dashboard/dataset/dataset_settings_viewconfig_tab.tsx

```diff
@@ -14,24 +14,89 @@ import {
   Slider,
   Divider,
 } from "antd";
-import * as React from "react";
+import React, { useMemo, useState } from "react";
 import { Vector3Input } from "libs/vector_input";
 import { validateLayerViewConfigurationObjectJSON, syncValidator } from "types/validation";
 import { getDefaultLayerViewConfiguration } from "types/schemas/dataset_view_configuration.schema";
-import {
+import messages, {
   RecommendedConfiguration,
   layerViewConfigurations,
   settings,
   settingsTooltips,
 } from "messages";
-import type { DatasetLayerConfiguration } from "oxalis/store";
+import type { DatasetConfiguration, DatasetLayerConfiguration } from "oxalis/store";
 import { FormItemWithInfo, jsonEditStyle } from "./helper_components";
 import { BLEND_MODES } from "oxalis/constants";
 import ColorLayerOrderingTable from "./color_layer_ordering_component";
+import { APIDatasetId } from "types/api_flow_types";
+import { getAgglomeratesForDatasetLayer, getMappingsForDatasetLayer } from "admin/admin_rest_api";
 
 const FormItem = Form.Item;
 
-export default function DatasetSettingsViewConfigTab() {
+export default function DatasetSettingsViewConfigTab(props: {
+  datasetId: APIDatasetId;
+  dataStoreURL: string | undefined;
+}) {
+  const { datasetId, dataStoreURL } = props;
+  const [availableMappingsPerLayerCache, setAvailableMappingsPerLayer] = useState(
```

Comment:
`useState<T>` accepts a type hint.

```suggestion
  const [availableMappingsPerLayerCache, setAvailableMappingsPerLayer] = useState<Record<string, [string[], string[]]>>(
```